sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -93,44 +94,58 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
       queryMethod: KalixMethod,
       queryInputSchemaDescriptor: Option[ProtoMessageDescriptors],
       queryOutputSchemaDescriptor: ProtoMessageDescriptors) = {
-      val annotatedMethods = RestServiceIntrospector
+
+      val annotatedQueryMethods = RestServiceIntrospector
         .inspectService(component)
         .methods
         .filter(_.javaMethod.getAnnotation(classOf[Query]) != null)
-      if (annotatedMethods.isEmpty)
-        throw new IllegalArgumentException(
-          s"No valid query method found in class ${component.getName}. " +
+
+      if (annotatedQueryMethods.isEmpty)
+        throw ServiceIntrospectionException(
+          component,
+          s"No valid query method found. " +
           "Views should have a method annotated with @Query and exposed by a REST annotation")
-      if (annotatedMethods.size > 1)
-        throw new IllegalArgumentException(
+
+      if (annotatedQueryMethods.size > 1)
+        throw ServiceIntrospectionException(
```

Comment:
Using `ServiceIntrospectionException` here and above because better to inform user what is wrong.  

See https://github.com/lightbend/kalix/issues/7801